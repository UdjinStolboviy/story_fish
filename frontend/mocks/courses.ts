export const response = {
  data: [
    {
      id: 1,
      attributes: {
        header: "Hands-On React. Build advanced React JS Frontend with expert",
        description:
          "**React is the most popular library for building frontend web applications**. Step-by-step by diving into all the basics, I'll introduce you to advanced concepts as well.\n\nWe'll build the minesweeper application from scratch:\n\n-   setup of the development environment\n\n-   configuration of the React JS app\n\n-   basic algorithms of Minesweeper\n\nStep-by-step we'll build the minesweeper game. **During the course, we'll cover the most important topics**.\n\nFirst will be a configuration of the ReactJS application with Typescript and using build tools Webpack+Babel.\n\nSecond, there're presented **TDD or TLD approaches** that I'll try to practice with you during the coding sessions. Probably you don't have enough experience with the tests, but it's ok, there's still a good way to learn it from the course. When you work with code and cover it with test cases, it provides you with guarantees that your code works as you expected. This is the purpose and benefit of the tests.\\\nTools for testing: **Jest, Stryker, React Testing Library**\n\nAlso, I'll try to cover basic **Typescript** concepts and show you a way to improve these skills. During the coding sessions, we'll cover advanced Javascript techniques to make sense of commonly used JS features.\n\nStorybook is the most popular way to build the components library. From our side, it's the most basic part of the course. We'll install and configure Storybook, which will provide us with a full components spec. This approach is called Components Driven Development.\n\n**CSS--IN-JS, emotion, and Styled-Components** are very powerful toolkits to build UI components. Even very complex components styles can be easily produced and supported by it.\n\nWe'll introduce you to **React Hooks** - it's the basic way to make dynamic UI for the applications.\n\nWe'll make a review of **React Router** v5 and also React Router v6.\\\nLast but not least it's **Redux**. Redux is a beautiful way to manage an application state. **We'll use the Redux-Toolkit library** - it simplifies the work with Redux.\\\n**Check out the full curriculum and free preview videos. Join the course risk-free with a 30-day money-back guarantee!**\\\nSee you on the course!",
        createdAt: "2022-04-19T15:19:20.964Z",
        updatedAt: "2022-05-10T11:50:17.393Z",
        publishedAt: "2022-04-19T15:19:43.376Z",
        link: "https://www.udemy.com/course/hands-on-reactjs/",
        subtitle:
          "Practical React with Typescript and Testing Library. React and Storybook. TDD with React Hooks and React + Redux Toolkit",
        cover: {
          data: {
            id: 3,
            attributes: {
              name: "cover4.png",
              alternativeText: "cover4.png",
              caption: "cover4.png",
              width: 1368,
              height: 770,
              formats: {
                thumbnail: {
                  name: "thumbnail_cover4.png",
                  hash: "thumbnail_cover4_dcf1d6a3dc",
                  ext: ".png",
                  mime: "image/png",
                  path: null,
                  width: 245,
                  height: 138,
                  size: 33.52,
                  url: "/covers/hands-on_reactjs_cover.png",
                },
                large: {
                  name: "large_cover4.png",
                  hash: "large_cover4_dcf1d6a3dc",
                  ext: ".png",
                  mime: "image/png",
                  path: null,
                  width: 1000,
                  height: 563,
                  size: 263.3,
                  url: "/covers/hands-on_reactjs_cover.png",
                },
                medium: {
                  name: "medium_cover4.png",
                  hash: "medium_cover4_dcf1d6a3dc",
                  ext: ".png",
                  mime: "image/png",
                  path: null,
                  width: 750,
                  height: 422,
                  size: 165.98,
                  url: "/covers/hands-on_reactjs_cover.png",
                },
                small: {
                  name: "small_cover4.png",
                  hash: "small_cover4_dcf1d6a3dc",
                  ext: ".png",
                  mime: "image/png",
                  path: null,
                  width: 500,
                  height: 281,
                  size: 89.37,
                  url: "/covers/hands-on_reactjs_cover.png",
                },
              },
              hash: "cover4_dcf1d6a3dc",
              ext: ".png",
              mime: "image/png",
              size: 73.37,
              url: "/covers/hands-on_reactjs_cover.png",
              previewUrl: null,
              provider: "local",
              provider_metadata: null,
              createdAt: "2022-05-09T11:05:18.318Z",
              updatedAt: "2022-05-09T11:05:18.318Z",
            },
          },
        },
      },
    },
    {
      id: 2,
      attributes: {
        header: "TypeScript Basic: Type System, Interfaces, Types, Generics",
        description:
          "**TypeScript** is a programming language developed and maintained by Microsoft. It is a strict syntactical superset of JavaScript and adds optional static typing to the language. TypeScript is designed for the development of large applications and transcompiles to JavaScript. As TypeScript is a superset of JavaScript, existing JavaScript programs are also valid TypeScript programs.\n\nTypeScript may be used to develop JavaScript applications for both client-side and server-side execution (as with Node.js or Deno). There are multiple options available for transcompilation. Either the default TypeScript Checker can be used, or the Babel compiler can be invoked to convert TypeScript to JavaScript.\n\nTypeScript is **JavaScript with syntax for types.**\n\nTypeScript is a strongly typed programming language that builds on JavaScript, giving you better tooling at any scale.\n\nOn the course we quickly create a basic project and then review the most common features\n\n**Who this course is for:**\n\n-   Professional Developers that are looking to keep current by learning the latest programming language of the web development world\n\n-   Backend Developers doing a transition to full stack development\n\n-   Frontend Developers that want to improve their frontend skills\n\n-   Javascript Developers in general that use node as a backend server and would like to learn how to write node in Typescript\n\n-   Javascript Developers looking to learn Typescript",
        createdAt: "2022-04-19T15:20:35.240Z",
        updatedAt: "2022-05-10T11:50:50.670Z",
        publishedAt: "2022-04-19T15:20:38.302Z",
        link: "https://www.udemy.com/course/typescript-in-a-nutshell/",
        subtitle:
          "Learn basic of Typescript, Run Typescript + Webpack and Babel in 2 hour",
        cover: {
          data: {
            id: 4,
            attributes: {
              name: "tslogo.jpg",
              alternativeText: "tslogo.jpg",
              caption: "tslogo.jpg",
              width: 1004,
              height: 462,
              formats: {
                thumbnail: {
                  name: "thumbnail_tslogo.jpg",
                  hash: "thumbnail_tslogo_91b6c1c0da",
                  ext: ".jpg",
                  mime: "image/jpeg",
                  path: null,
                  width: 245,
                  height: 113,
                  size: 3.7,
                  url: "/covers/ts_cover.jpg",
                },
                large: {
                  name: "large_tslogo.jpg",
                  hash: "large_tslogo_91b6c1c0da",
                  ext: ".jpg",
                  mime: "image/jpeg",
                  path: null,
                  width: 1000,
                  height: 460,
                  size: 43.97,
                  url: "/covers/ts_cover.jpg",
                },
                medium: {
                  name: "medium_tslogo.jpg",
                  hash: "medium_tslogo_91b6c1c0da",
                  ext: ".jpg",
                  mime: "image/jpeg",
                  path: null,
                  width: 750,
                  height: 345,
                  size: 25.76,
                  url: "/covers/ts_cover.jpg",
                },
                small: {
                  name: "small_tslogo.jpg",
                  hash: "small_tslogo_91b6c1c0da",
                  ext: ".jpg",
                  mime: "image/jpeg",
                  path: null,
                  width: 500,
                  height: 230,
                  size: 12.03,
                  url: "/covers/ts_cover.jpg",
                },
              },
              hash: "tslogo_91b6c1c0da",
              ext: ".jpg",
              mime: "image/jpeg",
              size: 44.39,
              url: "/covers/ts_cover.jpg",
              previewUrl: null,
              provider: "local",
              provider_metadata: null,
              createdAt: "2022-05-09T11:06:23.764Z",
              updatedAt: "2022-05-09T11:06:23.764Z",
            },
          },
        },
      },
    },
  ],
  meta: {
    pagination: {
      page: 1,
      pageSize: 25,
      pageCount: 1,
      total: 2,
    },
  },
};
