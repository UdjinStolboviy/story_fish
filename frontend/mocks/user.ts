export const mockUser = {
  jwt: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NSwiaWF0IjoxNjUwOTAxNjEwLCJleHAiOjE2NTM0OTM2MTB9.5XCgAR25c3AH_09Jv31CWNF-48mpRqGUat79XGYuNJo",
  user: {
    id: 5,
    username: "John Doe",
    email: "john@doe.com",
    password: "testpass123!",
    provider: "local",
    confirmed: true,
    blocked: false,
    createdAt: "2022-04-20T12:58:41.661Z",
    updatedAt: "2022-04-20T12:58:41.661Z",
  },
};

export const ValidationError = {
  error: {
    status: 400,
    name: "ValidationError",
    message: "Invalid identifier or password",
    details: {},
  },
};

export const RegistrationError = {
  error: {
    status: 400,
    name: "ApplicationError",
    message: "An error occurred during account creation",
    details: {},
  },
};
