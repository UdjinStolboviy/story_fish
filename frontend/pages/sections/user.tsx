import type { NextPage } from "next";
import { useRouter } from "next/router";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";

import { RootState, AppDispatch } from "@/store";
import { selectUser, logout } from "@/services/userSlice";

import { CenteredTile } from "@/components/Tile";
import { Button } from "@/components/Button";

const User: NextPage = () => {
  const router = useRouter();
  const { username, email, error } = useSelector<RootState, RootState["user"]>(
    selectUser
  );
  const dispatch = useDispatch<AppDispatch>();

  useEffect(() => {
    if (!username || Boolean(error)) {
      dispatch(logout());
      router.push("/login");
    }
  }, []);

  const logoutHandler = () => {
    dispatch(logout());
    router.push("/");
  };

  return username && email ? (
    <CenteredTile header="Profile">
      <h3>username: {username}</h3>
      <h3>email: {email}</h3>
      <Button onClick={logoutHandler}>Logout</Button>
    </CenteredTile>
  ) : null;
};

export default User;
