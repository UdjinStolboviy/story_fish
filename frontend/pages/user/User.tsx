import type { NextPage } from "next";
import { useRouter } from "next/router";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";

import { RootState, AppDispatch } from "@/store";
import { selectUser, logout, UserState } from "@/services/userSlice";

import { CenteredTile } from "@/components/Tile";
import { Button } from "@/components/Button";

import Head from "next/head";
import { Layout } from "@/components/Layout";

import { useState } from "react";
import { clearUserInfoFromLocalStorage } from "@/utils/utils";
import { useQueryClient } from "react-query";

const User: NextPage = () => {
  const router = useRouter();
  const [user, setUser] = useState<UserState | null>(null);
  const queryClient = useQueryClient();
  console.log("user ----------------  s ", user);
  useEffect(() => {
    const user: UserState | undefined = queryClient.getQueryData("user");

    if (!user) {
      router.push("/login");
    } else {
      setUser(user);
    }
  }, []);

  const logoutHandler = () => {
    clearUserInfoFromLocalStorage();
    router.push("/");
  };

  return (
    <>
      <Head>
        <title>CoursesBox</title>
        <meta name="description" content="IT courses for everyone" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Layout>
        {user && user.username && user.email ? (
          <CenteredTile header="Profile">
            <h3>username: {user.username}</h3>
            <h3>email: {user.email}</h3>
            <Button onClick={logoutHandler}>Logout</Button>
          </CenteredTile>
        ) : null}
      </Layout>
    </>
  );
};

export default User;
