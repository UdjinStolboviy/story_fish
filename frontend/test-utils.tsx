import { FC, ReactElement } from "react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@emotion/react";

import { rootReducer, RootState } from "./store";
import { Themes } from "./styles/themes";
import { Layout } from "./components/Layout";

type Options = { preloadedState?: RootState } & RenderOptions;

const customRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

const pageRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <Layout>{children}</Layout>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, pageRender };
