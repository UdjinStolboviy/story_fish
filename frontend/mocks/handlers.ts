import { rest } from "msw";

import { LoginData, RegistrationData } from "@/services/userSlice";

import { mockUser, ValidationError, RegistrationError } from "./user";

const api_url = process.env.NEXT_PUBLIC_STRAPI_API_URL;

export const handlers = [
  rest.post<LoginData>(`${api_url}/auth/local`, (req, res, ctx) => {
    const data = req.body;

    if (
      data.identifier === mockUser.user.email &&
      data.password === mockUser.user.password
    ) {
      return res(ctx.status(200), ctx.json(mockUser));
    }

    return res(ctx.status(400), ctx.json(ValidationError));
  }),

  rest.get(`${api_url}/users/me`, (req, res, ctx) => {
    if (req.headers.get("Authorization") === `Bearer ${mockUser.jwt}`) {
      return res(ctx.status(200), ctx.json(mockUser.user));
    }

    return res(ctx.status(400), ctx.json(ValidationError));
  }),

  rest.post<RegistrationData>(
    `${api_url}/auth/local/register`,
    (req, res, ctx) => {
      const data = req.body;

      if (
        data.email === mockUser.user.email &&
        data.password === mockUser.user.password &&
        data.username === mockUser.user.username
      ) {
        return res(ctx.status(200), ctx.json(mockUser));
      }

      return res(ctx.status(400), ctx.json(RegistrationError));
    }
  ),
];
