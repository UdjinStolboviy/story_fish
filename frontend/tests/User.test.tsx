import { useRouter } from "next/router";

import { pageRender as render, screen, waitFor } from "@/test-utils";
import { mockUser } from "@/mocks/user";
import User from "@/pages/user";

jest.mock("next/router", () => ({
  ...jest.requireActual("next/router"),
  useRouter: jest.fn().mockReturnValue({
    query: {},
    push: jest.fn(),
  }),
}));

const preloadedState = {
  user: {
    jwt: mockUser.jwt,
    email: mockUser.user.email,
    username: mockUser.user.username,
  },
};

describe("User page", () => {
  it("Render check", async () => {
    const { container } = render(<User />, {
      preloadedState,
    });
    expect(container).toMatchSnapshot();
  });
  it("Load user", async () => {
    const { container } = render(<User />, {
      preloadedState,
    });

    await screen.findByText("Profile");

    expect(
      screen.getByText(`username: ${mockUser.user.username}`)
    ).toBeInTheDocument();
    expect(
      screen.getByText(`email: ${mockUser.user.email}`)
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
  it("Redirect to login page", async () => {
    // Mock the router
    const push = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({
      query: {},
      push,
    });

    render(<User />, {
      preloadedState: {
        user: {
          jwt: "",
          email: "",
          username: "",
        },
      },
    });

    await waitFor(() => expect(push).toHaveBeenCalledWith("/login"));
  });
});
