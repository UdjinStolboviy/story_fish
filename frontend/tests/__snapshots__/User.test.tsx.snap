// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User page Load user 1`] = `
<div>
  <div
    class="css-u0jlnf"
  >
    <a
      class="css-4w3zwj"
      href="/"
    >
      <header
        class="css-rq0vky"
        size="3"
      >
        <span
          class="logo_short"
        >
          C8X
        </span>
        <span
          class="logo_full"
        >
          CoursesBox
        </span>
      </header>
    </a>
    <nav
      class="css-1sstk05"
    >
      <button
        class="css-15xqhy1"
        title="User"
      >
        <svg
          aria-label="User"
          class="css-z3g3v6"
          fill="currentColor"
          height="1rem"
          href="/user"
          role="img"
          viewBox="0 0 1024 1024"
          width="1rem"
        >
          <path
            d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
          />
        </svg>
      </button>
      <button
        class="css-15xqhy1"
        title="Moon"
      >
        <svg
          aria-label="Moon"
          class="css-z3g3v6"
          fill="currentColor"
          height="1rem"
          role="img"
          viewBox="0 0 24 24"
          width="1rem"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10 6a8 8 0 0011.955 6.956C21.474 18.03 17.2 22 12 22 6.477 22 2 17.523 2 12c0-5.2 3.97-9.474 9.044-9.955A7.963 7.963 0 0010 6zm-6 6a8 8 0 008 8 8.006 8.006 0 006.957-4.045c-.316.03-.636.045-.957.045-5.523 0-10-4.477-10-10 0-.321.015-.64.045-.957A8.006 8.006 0 004 12zm14.164-9.709L19 2.5v1l-.836.209a2 2 0 00-1.455 1.455L16.5 6h-1l-.209-.836a2 2 0 00-1.455-1.455L13 3.5v-1l.836-.209A2 2 0 0015.29.836L15.5 0h1l.209.836a2 2 0 001.455 1.455zm5 5L24 7.5v1l-.836.209a2 2 0 00-1.455 1.455L21.5 11h-1l-.209-.836a2 2 0 00-1.455-1.455L18 8.5v-1l.836-.209a2 2 0 001.455-1.455L20.5 5h1l.209.836a2 2 0 001.455 1.455z"
          />
        </svg>
      </button>
    </nav>
    <label
      class="css-fabfz"
      height="7"
      width="20"
    >
      <span
        class="css-1grri41"
      />
      <div
        class="css-52ovgc"
      >
        <input
          class="css-1bj11r8"
          placeholder="Search"
          value=""
        />
        <svg
          aria-hidden="true"
          aria-label="Search"
          class="css-mjc9ur"
          data-icon="search"
          fill="currentColor"
          focusable="false"
          height="2rem"
          role="img"
          viewBox="64 64 896 896"
          width="2rem"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </div>
      <span
        class="css-1reg724"
        role="alert"
      />
    </label>
    <main
      class="css-tee3o6"
    >
      <div
        class="css-qbqlvy"
      >
        <section
          class="css-6q0t2w"
        >
          <h2>
            Profile
          </h2>
          <h3>
            username: 
            John Doe
          </h3>
          <h3>
            email: 
            john@doe.com
          </h3>
          <button
            class="css-1rttvxo"
            color="primary"
          >
            Logout
          </button>
        </section>
      </div>
    </main>
    <footer
      class="css-1y0dtms"
    >
      © 
      2022
       NickOvchinnikov. All rights reserved.
    </footer>
  </div>
</div>
`;

exports[`User page Render check 1`] = `
<div>
  <div
    class="css-u0jlnf"
  >
    <a
      class="css-4w3zwj"
      href="/"
    >
      <header
        class="css-rq0vky"
        size="3"
      >
        <span
          class="logo_short"
        >
          C8X
        </span>
        <span
          class="logo_full"
        >
          CoursesBox
        </span>
      </header>
    </a>
    <nav
      class="css-1sstk05"
    >
      <button
        class="css-15xqhy1"
        title="User"
      >
        <svg
          aria-label="User"
          class="css-z3g3v6"
          fill="currentColor"
          height="1rem"
          href="/user"
          role="img"
          viewBox="0 0 1024 1024"
          width="1rem"
        >
          <path
            d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
          />
        </svg>
      </button>
      <button
        class="css-15xqhy1"
        title="Moon"
      >
        <svg
          aria-label="Moon"
          class="css-z3g3v6"
          fill="currentColor"
          height="1rem"
          role="img"
          viewBox="0 0 24 24"
          width="1rem"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10 6a8 8 0 0011.955 6.956C21.474 18.03 17.2 22 12 22 6.477 22 2 17.523 2 12c0-5.2 3.97-9.474 9.044-9.955A7.963 7.963 0 0010 6zm-6 6a8 8 0 008 8 8.006 8.006 0 006.957-4.045c-.316.03-.636.045-.957.045-5.523 0-10-4.477-10-10 0-.321.015-.64.045-.957A8.006 8.006 0 004 12zm14.164-9.709L19 2.5v1l-.836.209a2 2 0 00-1.455 1.455L16.5 6h-1l-.209-.836a2 2 0 00-1.455-1.455L13 3.5v-1l.836-.209A2 2 0 0015.29.836L15.5 0h1l.209.836a2 2 0 001.455 1.455zm5 5L24 7.5v1l-.836.209a2 2 0 00-1.455 1.455L21.5 11h-1l-.209-.836a2 2 0 00-1.455-1.455L18 8.5v-1l.836-.209a2 2 0 001.455-1.455L20.5 5h1l.209.836a2 2 0 001.455 1.455z"
          />
        </svg>
      </button>
    </nav>
    <label
      class="css-fabfz"
      height="7"
      width="20"
    >
      <span
        class="css-1grri41"
      />
      <div
        class="css-52ovgc"
      >
        <input
          class="css-1bj11r8"
          placeholder="Search"
          value=""
        />
        <svg
          aria-hidden="true"
          aria-label="Search"
          class="css-mjc9ur"
          data-icon="search"
          fill="currentColor"
          focusable="false"
          height="2rem"
          role="img"
          viewBox="64 64 896 896"
          width="2rem"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </div>
      <span
        class="css-1reg724"
        role="alert"
      />
    </label>
    <main
      class="css-tee3o6"
    >
      <div
        class="css-qbqlvy"
      >
        <section
          class="css-6q0t2w"
        >
          <h2>
            Profile
          </h2>
          <h3>
            username: 
            John Doe
          </h3>
          <h3>
            email: 
            john@doe.com
          </h3>
          <button
            class="css-1rttvxo"
            color="primary"
          >
            Logout
          </button>
        </section>
      </div>
    </main>
    <footer
      class="css-1y0dtms"
    >
      © 
      2022
       NickOvchinnikov. All rights reserved.
    </footer>
  </div>
</div>
`;
